_rules/inline-link-in-paragraph-has-distinguishable-border-36f116.md

```diff
@@ -0,0 +1,159 @@
+---
+id: 36f116
+name: Inline link has distinguishable border
+rule_type: atomic
+description: |
+  This rule checks that inline links have a border that distinguishes them from the surrounding text
+accessibility_requirements:
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+  previous authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], where the closest element (including itself) that generates an [inline box][] is part of a [line box][] that has [inline boxes][inline box] generated from an element containing [visible][] [text nodes][text node] that do not have a [semantic link][] as a [descendant][] in the [flat tree][].
+
+**Note:** The test targets are visible link elements that are part of a block of content that is displayed inline.
+
+## Expectation
+
+Each target element has a [visible][] [border](https://drafts.csswg.org/css-backgrounds/#borders).
+
+## Assumptions
+
+- This rule assumes that [`br` HTML elements][br] are only used for line breaks that are actually part of the content and not for separating thematic groups of the content. 
+- This rule assumes that any change in font is sufficiently distinguishable, and that fonts are loaded when they are present.
+- This rule assumes that if multiple colors are used in the visible text nodes of the block of content then color can not be a distinguishing factor.
+- This rule assumes that if `border` is used in the different visible text nodes of the block of content then `border` can not be a distinguishing factor.
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/WAI/WCAG21/Techniques/general/G183)
+- [F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73)
+- [C15: Using CSS to change the presentation of a user interface component when it receives focus](https://www.w3.org/WAI/WCAG21/Techniques/css/C15)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This is a link that is a descendant of a paragraph element, and therefore in an inline block of content. The link has a distinguishing bottom border in addition to being distinguishable by color.
+
+```html
+<style>
+	* {
+		color: black;
+	}
+	a.test {
+		color: blue;
+		text-decoration: none;
+		border-style: solid;
+		border-color: red;
+		border-width: 0px;
+		border-bottom-width: 1px;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+### Failed
```

Comment:
Because you're using "visible", I think you should include examples when the border has a color with an alpha of 0, and one where the border is clipped in some way.

Reply:
Do we really want to overload the rule with different ways to calculate the visibility of elements? 
I've added a further example with an alpha value making it transparent. I'll add further examples if people believe we need them.